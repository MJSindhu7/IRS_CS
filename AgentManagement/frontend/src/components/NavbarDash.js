import React, { Component } from 'react';
import logo from './../css/IRS.png'

class NavbarDash extends Component {
    
  handleLogout = () => {
    window.localStorage.clear();
  };

  render() {
    return (
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark mb-4 fixed-top">
        
          <img  src={logo}
           width="33"
           height="30"
           className="d-inline-block align-top"/>
           
          <a className="navbar-brand" href="/AgentDashboard">
            Dashboard
          </a>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
          >
            <span className="navbar-toggler-icon"></span>
          </button>

          <div className="collapse navbar-collapse" id="mobile-nav">
            
            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <p className="nav-link" >
                  Hi, {localStorage.getItem("userName")}
                </p>
              </li>
              <li className="nav-item">
                <a className="nav-link" href="/" onClick={this.handleLogout}>
                  Logout
                </a>
              </li>
            </ul>
          </div>
        
      </nav>
    );
  }
}

export default NavbarDash;