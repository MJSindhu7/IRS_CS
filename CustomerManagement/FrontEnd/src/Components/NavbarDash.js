import React, { Component } from 'react';
import {getName} from './utils.js'

class NavbarDash extends Component {
    
  handleLogout = () => {
    window.localStorage.clear();
  };

  render() {
    return (
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark mb-4">
        <div className="container">
          <a className="navbar-brand" href="/AgentDashboard">
            IRS Dashboard
          </a>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
          >
            <span className="navbar-toggler-icon"></span>
          </button>

          <div className="collapse navbar-collapse" id="mobile-nav">
            
            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <p className="nav-link" >
                   {getName()}
                </p>
              </li>
              <li className="nav-item">
                <a className="nav-link" href="/OrgOwnerLogin" onClick={this.handleLogout}>
                  Logout
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
}

export default NavbarDash;