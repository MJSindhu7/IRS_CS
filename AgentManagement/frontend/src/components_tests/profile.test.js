import React from 'react';
import renderer from 'react-test-renderer';
import Profile from '../components/profile';
import NavbarDash from "../components/NavbarDash";
import Sidebar from '../components/Sidebar'
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
configure({ adapter: new Adapter() });


describe('test cases in update profile',() =>{
    test('renders list-items, to display button', () => {
        const wrapper = shallow(<Profile />);
        expect(wrapper.find('UpdateButton')).toBeDefined();
      })
      it('should render NavbarDash component correctly', () => {
        const component = shallow(<NavbarDash />);
        expect(component).toMatchSnapshot();
      });
    it('should render Sidebar component correctly', () => {
        const component = shallow(<Sidebar />);
        expect(component).toMatchSnapshot();
      });
    });