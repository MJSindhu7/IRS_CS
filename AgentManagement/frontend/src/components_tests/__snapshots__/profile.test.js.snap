// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test case in update profile should render NavbarDash component correctly 1`] = `ShallowWrapper {}`;

exports[`test case in update profile should render Sidebar component correctly 1`] = `ShallowWrapper {}`;

exports[`test cases in update profile should render NavbarDash component correctly 1`] = `ShallowWrapper {}`;

exports[`test cases in update profile should render Sidebar component correctly 1`] = `ShallowWrapper {}`;
